corporate/lib/stripe.py

```diff
@@ -633,6 +633,19 @@ def do_change_remote_server_plan_type(remote_server: RemoteZulipServer, plan_typ
     )
 
 
+@transaction.atomic
+def do_deactivate_remote_server(remote_server: RemoteZulipServer) -> None:
+    if not remote_server.deactivated:
+        remote_server.deactivated = True
+        remote_server.save(update_fields=["deactivated"])
+        RemoteZulipServerAuditLog.objects.create(
+            event_type=RealmAuditLog.REMOTE_SERVER_DEACTIVATED_CHANGED,
+            server=remote_server,
+            event_time=timezone_now(),
+            extra_data=orjson.dumps({"old_value": False, "new_value": True}).decode(),
```

Comment:
I'm not sure we need `extra_data` here; look at our realm or stream deactivation code for inspiration.